docs/site/Routing-requests.md

```diff
@@ -0,0 +1,87 @@
+---
+lang: en
+title: 'Routing requests'
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Routing-requests.html
+---
+
+## Routing Requests
+
+This is an action in the default HTTP sequence. Its responsibility is to find a
+route that can handle a given http request. By default, the `FindRoute` action
+uses the `RoutingTable` from `@loopback/rest` to match requests against
+registered routes including controller methods using `request.method` and
+`request.path`. For example:
+
+- GET /orders => OrderController.getOrders (`@get('/orders')`)
+- GET /orders/123 => OrderController.getOrderById (`@get('/orders/{id}')`)
+- GET /orders/count => OrderController.getOrderCount (`@get('/orders/count')`)
+- POST /orders => OrderController.createOrder (`@post('/orders')`)
+
+## Customize the `FindRoute` action
+
+The `FindRoute` action is bound to `SequenceActions.FIND_ROUTE`
+('rest.sequence.actions.findRoute') and injected into the default sequence.
+
+To create your own `FindRoute` action, bind your implementation as follows:
+
+```ts
+const yourFindRoute: FindRoute = ...;
+app.bind(SequenceActions.FIND_ROUTE).to(yourFindRoute);
+```
+
+## Customize the REST Router
+
+Instead of rewriting `FindRoute` action completely, LoopBack 4 also allows you
+to simply replace the `RestRouter` implementation.
+
+The `@loopback/rest` module ships two built-in routers:
+
+- RegExpRouter: it keeps routes as an array and uses `path-to-regexp` to match
+  `request` to routes based on the path pattern
+- TrieRouter: it keeps routes as a `trie` tree and uses traversal to match
+  `request` to routes based on the hierarchy of the path
+
+For both routers, routes without variables are optimized in a map so that any
+requests matching to a fixed path can be resolved quickly.
+
+By default, `@loopback/rest` uses `RegExpRouter` as it performs well with up to
+a few hundred of routes. There is a simple benchmarking for `RegExpRouter` and
+`TrieRouter` at
+https://githhub.com/strongloop/loopback-next/benchmark/src/rest-routing/routing-table.ts.
+
+To change the router for REST routing, we can bind the router class as follows:
+
+```ts
+import {RestBindings} from '@loopback/rest';
+app.bind(RestBindings.ROUTER).toClass(TrieRouter);
```

Comment:
nitpick: Which file would this be in and where can users import `TrieRouter` from?

Reply:
Good catch.